src/foam/nanos/boot/Boot.java

```diff
@@ -113,17 +117,35 @@ public Object create(X x) {
       public void put(Object obj, Detachable sub) {
         NSpec sp = (NSpec) obj;
 
-        logger.info("Starting:", sp.getName());
+        logger.info("Invoking Service", sp.getName());
         root_.get(sp.getName());
       }
     });
 
     String startScript = System.getProperty("foam.main", "main");
     if ( startScript != null ) {
-      DAO    scriptDAO = (DAO) root_.get("scriptDAO");
+      DAO    scriptDAO = (DAO) root_.get("bootScriptDAO");
+      if ( scriptDAO == null ) {
+        logger.warning("DAO Not Found: bootScriptDAO. Falling back to scriptDAO");
+        scriptDAO = (DAO) root_.get("scriptDAO");
+      }
       Script script    = (Script) scriptDAO.find(startScript);
       if ( script != null ) {
-        ((Script) script.fclone()).runScript(root_);
+        logger.info("Boot,script", startScript);
+        script = (Script) script.fclone();
+        // REVIEW: want main to run in ReadOnlyContext, but tests to run normally
+        if ( "main".equals(startScript) ) {
+          try {
+            // NOTE: is read-only and will throw exception when it updates rundate.
+            script.runScript(new foam.core.ReadOnlyDAOContext(root_));
```

Comment:
Why is it ReadOnly?

Reply:
I don't want it to save back into the journals/bootscript file.  The bootscript is just that, boot only, can't have side effects for clustering.

Reply:
But we have other bootscripts when we want to run unit tests from the command-line, like the "TestReporter" script. Will this break scripts like those?

Reply:
no, only 'main' is run in a ReadOnlyDAOContext.

Reply:
Removed. 
This readonly restriction on main has been in the medusa branch a long time. Whatever it was protecting is no longer an issue.